### PR TITLE
fixes + no opencl in transpier.jl, etc

### PR DESCRIPTION
**Pull request by SimonDanisch**

### CODE REVIEW

**Comment by SimonDanisch** [comment]
@maleadt, what do I need to do about:
```
ERROR: LoadError: LoadError: Unable to load dependency file /var/lib/buildbot/workers/julia/GPUArrays-julia06-x86-64bit/packages/v0.6/CUDArt/deps/ext.jl.
Please run Pkg.build("CUDAdrv") and restart Julia.
```

**Comment by maleadt** [comment]
As you can see it is already an error during package build:

```
===============================[ ERROR: CUDArt ]================================
LoadError: Could not find NVIDIA CUDA compiler (nvcc); consider setting the CUDA_PATH environment variable.
while loading /var/lib/buildbot/workers/julia/GPUArrays-julia06-x86-64bit/packages/v0.6/CUDArt/deps/build.jl, in expression starting on line 388
```

yet CUDAdrv itself builds successfully on the very same bot: http://ci.maleadt.net:8010/#/builders/13/builds/38

Do you do something funky with `CUDA_*` env vars?

**Comment by SimonDanisch** [comment]
Not really... Doesn't it also happen before I even could mess with anything?

**Comment by maleadt** [comment]
OK found it , will fix in a second.

**Comment by maleadt** [comment]
CI issue fixed.

**Comment by SimonDanisch** [comment]
Great will try it. Thanks a lot!

File: src/backends/cudanative/cudanative.jl [hunk 1]
```diff
@@ -146,7 +146,7 @@ immutable CUFunction{T}
     kernel::T
 end
 
-if success(`nvcc --version`)
+if try success(`nvcc --version`); catch false; end
```
Comment (maleadt) [inline src/backends/cudanative/cudanative.jl]: `nvcc` is properly detected by CUDArt, so why not use it from there?
Also, CUDArt doesn't build without it, so it should be unconditionally available (if CUDArt has been built).

Comment (SimonDanisch) [reply src/backends/cudanative/cudanative.jl]: Ah good to know! I will need to rewrite this part for your compilation PR anyways.

**Pull request merged**